File: stepik-java-api/src/main/java/org/stepik/api/queries/courses/StepikCoursesGetQuery.java
package org.stepik.api.queries.courses;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.actions.StepikAbstractAction;
import org.stepik.api.objects.courses.Courses;
import org.stepik.api.queries.StepikAbstractGetQuery;
import org.stepik.api.urls.Urls;

/**
 * @author meanmail
 */
public class StepikCoursesGetQuery extends StepikAbstractGetQuery<StepikCoursesGetQuery, Courses> {
    public StepikCoursesGetQuery(StepikAbstractAction stepikAction) {
        super(stepikAction, Courses.class);
    }

    @NotNull
    public StepikCoursesGetQuery page(int page) {
        addParam("page", page);
        return this;
    }

    @NotNull
    public StepikCoursesGetQuery isFeatured(boolean value) {
        addParam("is_featured", value);
        return this;
    }

    @NotNull
    public StepikCoursesGetQuery tag(int value) {
        addParam("tag", value);
        return this;
    }

    @NotNull
    public StepikCoursesGetQuery language(@NotNull String value) {
        addParam("language", value);
        return this;
    }

    @NotNull
    public StepikCoursesGetQuery owner(int value) {
        addParam("owner", value);
        return this;
    }

    @NotNull
    public StepikCoursesGetQuery isIdeaCompatible(boolean value) {
        addParam("is_idea_compatible", value);
        return this;
    }

    @NotNull
    public StepikCoursesGetQuery enrolled(boolean value) {
        addParam("enrolled", value);
        return this;
    }

    @NotNull
    @Override
    protected String getUrl() {
        return Urls.COURSES;
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/queries/courses/StepikCoursesGetQuery.kt
package org.stepik.api.queries.courses

import org.stepik.api.actions.StepikAbstractAction
import org.stepik.api.objects.courses.Courses
import org.stepik.api.queries.StepikAbstractGetQuery

class StepikCoursesGetQuery(stepikAction: StepikAbstractAction) :
        StepikAbstractGetQuery<StepikCoursesGetQuery, Courses>(stepikAction, Courses::class.java) {
    
    fun page(page: Int): StepikCoursesGetQuery {
        addParam("page", page)
        return this
    }
    
    fun isFeatured(value: Boolean): StepikCoursesGetQuery {
        addParam("is_featured", value)
        return this
    }
    
    fun tag(value: Int): StepikCoursesGetQuery {
        addParam("tag", value)
        return this
    }
    
    fun language(value: String): StepikCoursesGetQuery {
        addParam("language", value)
        return this
    }
    
    fun owner(value: Int): StepikCoursesGetQuery {
        addParam("owner", value)
        return this
    }
    
    fun isIdeaCompatible(value: Boolean): StepikCoursesGetQuery {
        addParam("is_idea_compatible", value)
        return this
    }
    
    fun enrolled(value: Boolean): StepikCoursesGetQuery {
        addParam("enrolled", value)
        return this
    }
    
    override val url = "${stepikAction.stepikApiClient.host}/api/courses"
    
}
